src/coreclr/jit/lowerxarch.cpp

```diff
@@ -3536,13 +3545,15 @@ void Lowering::LowerHWIntrinsicGetElement(GenTreeHWIntrinsic* node)
 
     if ((simdBaseType == TYP_BYTE) || (simdBaseType == TYP_SHORT))
     {
-        // The intrinsic zeros the upper bits, so we need an explicit
+        // The extract intrinsics zero the upper bits, so we need an explicit
         // cast to ensure the result is properly sign extended
 
         LIR::Use use;
-        bool     foundUse = BlockRange().TryGetUse(node, &use);
 
-        GenTreeCast* cast = comp->gtNewCastNode(TYP_INT, node, /* isUnsigned */ true, simdBaseType);
+        bool foundUse     = BlockRange().TryGetUse(node, &use);
+        bool fromUnsigned = false;
+
+        GenTreeCast* cast = comp->gtNewCastNode(TYP_INT, node, fromUnsigned, simdBaseType);
```

Comment:
`fromUnsigned` is `false` here?

Reply:
Yes. `pextrb`/`pextrw` already zero-extend to the upper bits. So we only need to insert a cast for sign-extension (e.g. when the source type is TYP_BYTE or TYP_SHORT)

Reply:
There's short comment a couple lines up that explains why its just for those and not for all small types.